### PR TITLE
`Development`: Remove quiz pool REST call on exam detail page

### PR DESCRIPTION
**Pull request by rriyaldhi**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [ ] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [ ] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [ ] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [ ] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context

Currently, there is an API call to retrieve the quiz pool on the exam detail page, as shown below:

image

The quiz pool is retrieved to get the maximum points for the configured quiz questions on the quiz pool. The maximum points are then displayed on the checklist, as shown below:

image

This API call is not necessary as the maximum points of a quiz exam can be returned as part of the response of the exam details API call.

### Description

**Server**

- Add transient quizExamMaxPoints property to Exam entity.
- Refactor QuizPoolService.findByExamId into two different methods: findWithQuizQuestionsByExamId and findByExamId. The findByExamId is used by the GET exam details API to get the maxPoints of the QuizPool.
- Modify QuizPool GET API to return null instead of throwing 404 if quiz pool is not found.

**Client**

- Remove Quiz Pool API call on ExamChecklistComponent.
- Pass Exam.quizExamMaxPoints instead of QuizPool.maxPoints to ExamChecklistExerciseGroupTableComponent
- Rename quizPoolMaxPoints property of ExamChecklistExerciseGroupTableComponent to quizExamMaxPoints for consistency
- Modify QuizPoolComponent to create a new QuizPool object when the GET API returns null.

### Steps for Testing

Prerequisites:
- 1 Instructor (artemis_test_user_[11-15])
- 1 Exam with Quiz Pool:
  - **TS2**: [https://artemis-test2.artemis.cit.tum.de/course-management/22/exams/21](https://artemis-test2.artemis.cit.tum.de/course-management/22/exams/21)
  - **TS3**: [https://artemis-test3.artemis.cit.tum.de/course-management/25/exams/19](https://artemis-test3.artemis.cit.tum.de/course-management/25/exams/19)
  - **TS5**: [https://artemis-test5.artemis.cit.tum.de/course-management/15/exams/18](https://artemis-test5.artemis.cit.tum.de/course-management/15/exams/18)
  - **TS6**: [https://artemis-test6.artemis.cit.tum.de/course-management/24/exams/5](https://artemis-test6.artemis.cit.tum.de/course-management/24/exams/5)
- 1 Exam without Quiz Pool

1. Log in to Artemis.
2. Open the Network Tab on the Browser.
3. Navigate to the Exam Detail page with Quiz Pool.
4. Verify that there is no additional call to retrieve QuizPool and that the Quiz Exam section is displayed.

image

5. Navigate to the Exam Detail page without Quiz Pool.
6. Verify that there is no additional call to retrieve QuizPool.

image

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| Exam.java | 92% | ✅                           |
| QuizPoolService.java | 97% | ✅                           |
| ExamService.java | 93% | ✅                           |
| ExamResource.java | 95% | ✅                           |
| QuizPoolResource.java | 100% | ✅                           |
| exam-checklist-exercisegroup-table.component.ts | 95.34% | ✅              |
| exam-checklist.component.ts | 95.83% | ✅              |
| quiz-pool.component.ts | 93.33% | ✅              |

### CODE REVIEW

**Review by rstief — DISMISSED** [review]
tested on ts2, worked as expected.

**Review by RY997 — DISMISSED** [review]
Tested on TS2, worked well

File: src/main/java/de/tum/in/www1/artemis/web/rest/QuizPoolResource.java [hunk 1]
```diff
@@ -82,8 +85,12 @@ public ResponseEntity<QuizPool> getQuizPool(@PathVariable Long courseId, @PathVa
         log.info("REST request to get QuizPool given examId : {}", examId);
 
         validateCourseRole(courseId);
-        QuizPool quizPool = quizPoolService.findByExamId(examId);
+        Optional<QuizPool> quizPoolOptional = quizPoolService.findWithQuizQuestionsByExamId(examId);
+        if (quizPoolOptional.isEmpty()) {
+            throw new EntityNotFoundException(ENTITY_NAME, "examId=" + examId);
```
[the remark this answers is not in the document]
Comment (rriyaldhi) [reply src/main/java/de/tum/in/www1/artemis/web/rest/QuizPoolResource.java]: The EntityNotFoundException throw is removed.

**Review by Kroko-fant — APPROVED** [review]
Tested in a testing session worked as expected.

**Review by basak-akan — DISMISSED** [review]
Tested during the testing session on TS3, worked as described

**Review by sarpsahinalp — APPROVED** [review]
Tested in a testing session worked as expected.

**Review by Predixx — DISMISSED** [review]
tested on ts6 and worked for me.

**Review by LenaKahle — APPROVED** [review]
Approve, tested on ts2.

**Review by basak-akan — APPROVED** [review]
Retested on TS2 still works as expected

**Review by lennart-keller — APPROVED** [review]
Tested in testing session. Code also lgtm

**Pull request merged**